### PR TITLE
feat: UI show the user last search queries

### LINKED ISSUE #1855: UX (search): frontend to show the user's last search queries

### PR DESCRIPTION
**Pull request by andre-code**
PR to add the last queries searched by the user as suggestions in the search field.
Displays the last 5 searches, only for logged users.

last searches

Closes #1855 

/deploy #persist

### CODE REVIEW

File: client/src/utils/components/quicknav/KgLastQueriesApi.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+interface KgLastSearchResult {
+  queries?: string[],
+  error?: string,
+}
+export const TOTAL_QUERIES = 5;
+
+
+export const kgLastQueriesApi = createApi({
```
Comment (ciyer) [inline client/src/utils/components/quicknav/KgLastQueriesApi.ts:27]: I like this approach a lot! The question is, though, what is the right scope for this RTK-Query API?

We also have the "recent-projects" API that is quite similar, and I think it makes sense to combine that (and other future endpoints) into a `recentUserActivityApi`. 

I don't think the other query needs to integrated as part of this PR, but we can prepare for it by using the name and setting the `baseQuery` to `/ui-server/api/`.

File: client/src/utils/components/quicknav/KgLastQueriesApi.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+interface KgLastSearchResult {
+  queries?: string[],
+  error?: string,
+}
+export const TOTAL_QUERIES = 5;
```
Comment (ciyer) [inline client/src/utils/components/quicknav/KgLastQueriesApi.ts:24]: Do we store a total of 4 or 5 queries? I never see more than 4:

image

**Review by ciyer — CHANGES_REQUESTED** [review]
This is a cool feature! 🎉  

I have a comment about how the API is structured and one small thing about the number of search results that are shown, but they should both be easy to address.

File: client/src/utils/components/quicknav/KgLastQueriesApi.ts (hunk 1 above)
Comment (andre-code) [reply client/src/utils/components/quicknav/KgLastQueriesApi.ts]: Thanks, great suggestion. I moved the kgLastQueries to recentUserActivityApi.

File: client/src/utils/components/quicknav/KgLastQueriesApi.ts (hunk 2 above)
Comment (andre-code) [reply client/src/utils/components/quicknav/KgLastQueriesApi.ts]: The reason was that the last searches query also returns `*` and in the frontend we removed that option reducing the amount of results. I have added a change to always have the last 5 words if those exist.

**Review by ciyer — APPROVED** [review]
This is very nice to use!

**Pull request merged**